app/apps/handlers.js

```diff
@@ -1,122 +1,100 @@
-const ensureLoggedIn = require('connect-ensure-login').ensureLoggedIn;
+const { ensureLoggedIn } = require('connect-ensure-login');
+const { App, Bucket, User } = require('../api-client');
 
-const api = require('../api-client');
-const routes = require('../routes');
 
-
-exports.new_app = [
+exports.new = [
   ensureLoggedIn('/login'),
-  function (req, res, next) {
-    api.list_buckets()
-      .then((buckets_response) => {
-        const template_args = {
+  (req, res, next) => {
+
+    Bucket.list()
+
+      .then((buckets) => {
+        res.render('apps/new.html', {
           prefix: process.env.ENV + '-',
-          buckets: buckets_response.results,
-        };
-        res.render('apps/new.html', template_args);
+          buckets: buckets,
+        });
       })
+
       .catch(next);
   }
 ];
 
 
-exports.create_app = [
+exports.create = [
   ensureLoggedIn('/login'),
-  function (req, res, next) {
-    const app = {
+  (req, res, next) => {
+
+    const { url_for } = require('../routes');
+
+    new App({
       name: req.body.name,
       description: req.body.description,
       repo_url: req.body.repo_url,
       userapps: [],
-    };
+    })
 
```

Comment:
I'm not so sure about these spaces between these statements to be honest as it may look like they're not related to each other.

Reply:
Yep, I will remove them. It's in the [code standard](https://github.com/airbnb/javascript#whitespace--padded-blocks) too 🙈